modules/tinymce/src/core/test/ts/browser/DragDropOverridesTest.ts

```diff
@@ -30,14 +41,61 @@ describe('browser.tinymce.core.DragDropOverridesTest', () => {
       Mouse.mouseMoveTo(SugarBody.body(), 0, 0);
     });
 
+    const findEvent = (eventType: string): Optional<DragEvent> => Arr.find(events, ({ type }) => type === eventType);
+
+    const getDataTransferFromEvent = (eventType: string): Optional<DataTransfer> =>
+      findEvent(eventType).bind((event) => Optional.from(event.dataTransfer));
+
     const assertEventsDispatched = (expectedTypes: string[]) => {
       const eventTypes = Arr.map(events, (e) => e.type);
 
       assert.deepEqual(eventTypes, expectedTypes);
     };
 
-    const assertDndEvent = (expectedType: string, expectedClass: string, assertMouseCords = true) => {
-      const event = Arr.find(events, ({ type }) => type === expectedType).getOrDie(`Could not find expected event type: ${expectedType}`);
+    const assertDndEventDataTransfer = (eventType: string, spec: DataTransferSpec) =>
+      getDataTransferFromEvent(eventType).fold(
+        () => assert.fail(`Expected ${eventType} event to have dataTransfer object`),
+        (dataTransfer) => {
+          if (eventType === 'dragstart') {
+            assert.isTrue(DataTransferMode.isInReadWriteMode(dataTransfer), `Expected dataTransfer of ${eventType} to be in readwrite mode`);
+          } else if (eventType === 'drop') {
+            assert.isTrue(DataTransferMode.isInReadOnlyMode(dataTransfer), `Expected dataTransfer of ${eventType} event to be in readonly mode`);
+          } else {
+            assert.isTrue(DataTransferMode.isInProtectedMode(dataTransfer), `Expected dataTransfer of ${eventType} event to be in protected mode`);
+            // Temporarily set to read-only to allow checking of data
+            DataTransferMode.setReadOnlyMode(dataTransfer);
+          }
+
+          Arr.each(spec.data, ({ type, value }) => assert.equal(dataTransfer.getData(type), value, `Expected dataTransfer on "${eventType}" event to have ${type} data`));
+          assert.equal(dataTransfer.dropEffect, spec.dropEffect, `Expected dataTransfer on "${eventType}" event to have dropEffect`);
+          assert.equal(dataTransfer.effectAllowed, spec.effectAllowed, `Expected dataTransfer on "${eventType}" event to have effectAllowed`);
+          KAssert.eqOptional(`Expected dataTransfer on "${eventType}" event to have dragImage`, Optional.from(spec.dragImage), DataTransfer.getDragImage(dataTransfer));
+
+          const dtFiles = dataTransfer.files;
+          if (spec.files.length === 0) {
+            assert.isNull(dtFiles.item(0), `Expected dataTransfer on "${eventType}" event to have no files`);
+          } else {
+            Arr.each(spec.files, (specFile) => {
+              let fileExists = false;
+              for (let i = 0, l = dtFiles.length; i < l; ++i) {
```

Comment:
dtFiles is an array like so you should be able to use use Arr.each on that as well. But since this checks for a file if it exists `Arr.find` might be better not sure.

Reply:
I was originally intending to use the `item` method, which was the reason for the `for` loop. But since the `files` property is designed to be iterated anyway I can refactor it using `Arr.find`.